app/browser/tabs.js

```diff
@@ -85,6 +85,13 @@ const getPartition = (createProperties) => {
   return partition
 }
 
+const needsPartitionAssigned = (createProperties) => {
+  return !createProperties.openerTabId ||
+     (createProperties.isPrivate && createProperties.partition === undefined) ||
+     (createProperties.isPartitioned && createProperties.partition === undefined) ||
+     (createProperties.partitionNumber && createProperties.partition === undefined)
```

Comment:
I think this should be:
```
!createProperties.openerTabId ||
createProperties.isPrivate ||
createProperties.isPartitioned ||
createProperties.partitionNumber ||
createProperties.partition

In particular the last one would also be needed in case someone is creating a tab inside a particular partition and we also want the parent partition to be filled correctly.